playbooks/roles/caasp4/tasks/cleanup_nodes.yml

```diff
@@ -19,6 +19,11 @@
 # stack in setup_nodes.yml so we need to remove the resources again from
 # terraform state and let heat handle the reource cleanup
 
+- name: Does the terraform workspace exist
+  stat:
+    path: skuba_ci_terraform_workspace
+  register: _terraform_workspace
+
 - name: Does the terraform state exist
   stat:
     path: "{{ skuba_ci_terraform_workspace }}/terraform.tfstate"
```

Comment:
Could we use this check instead of adding a new one?

Reply:
is it possible that the tfstate file doesnt exist but the dir does? In that case, the workspace wont be totally cleaned up no?

Reply:
It's possible that the dir exists but tfstate not.

Reply:
in that case, we cannot use the tfstate for that as it would leave the dir around. Unless we dont care about that and only care about the tfstate file, in which case the dir cleanup can be dropped

Reply:
ok